Kitodo/src/main/java/org/kitodo/services/data/ImportService.java

```diff
@@ -0,0 +1,197 @@
+            logger.error(e.getLocalizedMessage());
+            throw new IllegalArgumentException("Error: OPAC '" + opac + "' is not supported!");
+        }
+    }
+
+    /**
+     * Load catalog names from OPAC configuration file and return them as a list of Strings.
+     *
+     * @return list of catalog names
+     * @throws IllegalArgumentException thrown if no catalogs are configured in OPAC configuration file
+     */
+    public List<String> getAvailableCatalogs() throws IllegalArgumentException {
+        try {
+            return OPACConfig.getCatalogs();
+        } catch (IllegalArgumentException e) {
+            logger.error(e.getLocalizedMessage());
+            throw new IllegalArgumentException("Error: no supported OPACs found in configuration file!");
+        }
+    }
+
+    /**
+     * Get the full record with the given ID from the catalog.
+     *
+     * @param opac The ID of the catalog that will be queried.
+     * @param id   The ID of the record that will be imported.
+     * @return The queried record transformed into Kitodo internal format.
+     */
+    public Document getSelectedRecord(String opac, String id) {
+        importModule = initializeImportModule();
+        Document sruResponse = importModule.getFullRecordById(opac, id);
+        File xsltFile = new File(ConfigCore.getParameter(ParameterCore.DIR_XSLT) + OPACConfig.getXsltMappingFile(opac));
+        return transformXmlByXslt(convertDocumentToString(sruResponse), xsltFile);
+    }
+
+    private Document transformXmlByXslt(String xmlString, File stylesheetFile) {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        factory.setNamespaceAware(true);
+        try {
+            SAXBuilder saxBuilder = new SAXBuilder();
+            DOMOutputter outputter = new DOMOutputter();
+            StreamSource transformSource = new StreamSource(stylesheetFile);
+            TransformerFactoryImpl transformerFactoryImpl = new TransformerFactoryImpl();
+            File outputFile = File.createTempFile("transformed", "xml");
+            try (FileOutputStream outputStream = new FileOutputStream(outputFile)) {
+                Transformer xsltTransformer = transformerFactoryImpl.newTransformer(transformSource);
+                TransformerHandler handler = ((SAXTransformerFactory) SAXTransformerFactory.newInstance()).newTransformerHandler();
+                handler.setResult(new StreamResult(outputStream));
+                Result saxResult = new SAXResult(handler);
+                SAXSource saxSource = new SAXSource(new InputSource(new StringReader(xmlString)));
+                xsltTransformer.transform(saxSource, saxResult);
+            }
+            return outputter.output(saxBuilder.build(outputFile));
+        } catch (JDOMException | IOException | TransformerException e) {
+            logger.error("Error in transforming the response in intern format : ", e);
+            return null;
```

Comment:
Null returns should be avoided. Is it possible to return here for e.g. empty Document?

Reply:
Done.